# Tanybase
Tanybase - Combined and Curated Tanycyte Database
Tanycytes are ependymal-like glial cells that play a crucial role in regulating the neuroendocrine system, metabolism, and glucose homeostasis. Tanycytes lie in the wall of the third ventricle and coordinate various functions. Traditionally, tanycytes were classified into four subtypes based on their position in the ventricular wall: alpha 1, alpha 2, beta 1 and beta 2 tanycytes. Recent scRNA-seq studies have investigated the role of tanycytes at the transcriptome level, but they fall short due to the limited number of tanycytic cells in the datasets. To address this shortcoming, two prominent databases, Tany-seq and Hypomap, were compiled by combining leading hypothalamus and tanycyte scRNA-seq studies on this topic. However, the number of tanycytes for each subtype and condition was still limited, even after these noble works. To eliminate this pitfall and help future researchers in their quest to identify the role of tanycytes to a greater extent, we combined these two databases into a single mouse tanycyte-specific database called tanybase. Tanybase have a total of 20610 cells from 4 tanycytic subtypes, 5 different diets and 3 different ages. 

In short, tanybase is formed by combining two scRNAseq databases of mouse tanycytes, Hypomap and Tany-seq. Hypomap and Tany-seq databases were combined using shared parameters, including diet, age, tanycyte subclusters, original identity and dataset. Unique cells in each database were combined, and cells with more than 15% mitochondrial transcripts, a total RNA count below 600, and fewer than 200 unique features were removed. We included only cells that were labelled as tanycyte according to the Allen Institute MapMyCells software (RRID:SCR_024672). For integration, tanybase was split into different layers for each dataset containing more than 100 cells and datasets that have less than 100 cells were combined (‘others’). Tanybase was scaled, normalized and variable features were determined using Seurat functions. Principal-component analysis was performed as well as integration for batch correction using canonical correlation analysis. After the integration, layers were re-joined. For dimensionality reduction, we performed UMAP clustering with FindClusters and FindNeighbors functions. Labels were transferred from corresponding databases; unlabelled cells were labelled using module scores and the markers provided in Tany-seq.

![Tany_UMAP_mapped](https://github.com/user-attachments/assets/ae4f8358-aebf-4bc0-b83e-431592ad7182)
